### DISCUSSION TITLE
Preserve invoked services (machines) after reaching final states?

### DISCUSSION THREAD

**Discussion by geekox86** [statelyai/xstate#3321] [Q&A]
Hello everyone.

I know this will sound like a use case for actors but I am asking this question because I find it more convenient to work with invoked services (namely for the easier lifecycle management). I just want to keep them under the parent's children services even after reaching final states. Upon invoking them again they can be replaced. Is there such a flag to achieve this behavior?

Thank you!

**Comment by davidkpiano** [comment]
Can you explain your use-case a bit more?

**Reply by geekox86** [reply]
I have a multi-step form where each step has its own service. The form is the parent machine that invokes the next step's service when the previous one is done. However, I would like to keep the service states after being done so that the user can go back to these steps and edit. Currently, I have two options:
1- make final state as non-final but then I will need to communicate on done data via sendParent
2- use actors instead of invoked services

My suggestion is to simply allow to keep the services (e.g. via a flag) even after reaching final states without being responsive to further events. They can be reinvoked and replaced. This way I can simply use on done data instead of sendParent. It is not that important though :)
